src/Symfony/Component/HttpFoundation/RedirectResponse.php

```diff
@@ -76,6 +83,10 @@ public function getTargetUrl()
      */
     public function setTargetUrl($url)
     {
+        if (!is_string($url)) {
```

Comment:
the doc comment is fine to me, we only accept string - and we work around the fact that the engine cannot do any cast for us until php7 by doing the cast. But the cast should be done inside the function, not when calling it as you did in that commit.
Note that I'm still not sure we should do this here. Can't you do the cast on *your* side?